From 8dda9436fdf1966079d57bf371787311627f10e8 Mon Sep 17 00:00:00 2001
From: Jackie264 <125248570+Jackie264@users.noreply.github.com>
Date: Sun, 31 May 2026 18:32:03 +0800
Subject: [PATCH] Update Makefile

---
 Makefile | 23 ++++++++++++++++++++---
 1 file changed, 20 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index a48c04d..2844434 100644
--- a/Makefile
+++ b/Makefile
@@ -7,10 +7,27 @@ LUCI_MAINTAINER:=George Sapkin <george@sapk.in>
 PKG_LICENSE:=GPL-2.0-only
 
 LUCI_TITLE:=LuCI support for AdGuard Home
-LUCI_DEPENDS:=+adguardhome +luci-base
-LUCI_EXTRA_DEPENDS:=adguardhome (>=0.107.73-r3)
+LUCI_DEPENDS:=+luci-base
 LUCI_PKGARCH:=all
+USERID:=adguardhome=853:adguardhome=853
 
-include ../../luci.mk
+PKG_UNPACK:=$(CURDIR)/.prepare.sh $(PKG_NAME) $(CURDIR) $(PKG_BUILD_DIR)
+
+define Package/luci-app-adguardhome/conffiles
+/etc/adguardhome/adguardhome.yaml
+/etc/config/adguardhome
+endef
+
+define Package/luci-app-adguardhome/prerm
+#!/bin/sh
+[ -n "$${IPKG_INSTROOT}" ] || { 
+    rm -f /tmp/luci-indexcache.*
+    rm -rf /tmp/luci-modulecache/
+    /etc/init.d/rpcd reload 2>/dev/null
+}
+exit 0
+endef
+
+include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature
